include/deal.II/distributed/solution_transfer.h

```diff
@@ -383,12 +383,27 @@ namespace parallel
       register_data_attach();
     };
 
-
-  } // namespace distributed
+    namespace DoFHandlerTypeLegacy
+    {
+      /**
+       * @deprecated Use dealii::parallel::distributed::SolutionTransfer
+       * without the DoFHandlerType template instead.
+       */
+      template <int dim,
+                typename VectorType,
+                typename DoFHandlerType = DoFHandler<dim>>
+      class DEAL_II_DEPRECATED SolutionTransfer
+        : public dealii::parallel::distributed::
+            SolutionTransfer<dim, VectorType, DoFHandlerType>
+      {
+        using dealii::parallel::distributed::
+          SolutionTransfer<dim, VectorType, DoFHandlerType>::SolutionTransfer;
```

Comment:
I am a bit confused why `DoFHandlerType` is passed to the non-legacy version. I guess remove `DoFHandlerType`  there will the done follow-up PRs?